HIP/hip-514-missing-hts-precompile-functions.md

```diff
@@ -0,0 +1,244 @@
+
+The Hedera Smart Contract Service (HSCS) aims to provide developers with the ability to manage tokens via smart contracts.
+Currently, capability is limited to create, transfer, mint, burn, associate, and dissociate actions.
+Whereas, HTS provides additional capabilities including update, pause, kyc, freeze and info retrieval.
+This means smart contract developer cannot capitalize on valuable HTS supported scenarios and can therefore not expose these in their contracts and DApps.
+
+To allow for more rich smart contract scenarios, these limitations on the HSCS should be addressed by providing capability parity where possible with HTS.
+
+## User stories
+
+As a smart contract developer, I would like to retreive fungible token metadata details through a precompile contract.
+
+As a smart contract developer, I would like to retreive non-fungible token metadata details through a precompile contract.
+
+As a smart contract developer, I would like to modify a tokens entity properties through a precompile contract.
+
+As a smart contract developer, I would like to modify a tokens KYC permissions through a precompile contract.
+
+As a smart contract developer, I would like to retreive a tokens KYC status through a precompile contract.
+
+As a smart contract developer, I would like to modify a tokens Freeze state through a precompile contract.
+
+As a smart contract developer, I would like to retreive a tokens Freeze status through a precompile contract.
+
+As a smart contract developer, I would like to modify a tokens Pause state through a precompile contract.
+
+As a smart contract developer, I would like to retreive a tokens Pause status through a precompile contract.
+
+As a smart contract developer, I would like to retrieve a tokens fee properties through a precompile contract.
+
+As a smart contract developer, I would like to retrieve a tokens key properties through a precompile contract.
+
+As a smart contract developer, I would like to delete a token through a precompile contract.
+
+As a smart contract developer, I would like to wipe an amount of fungible tokens from an account balance through a precompile contract.
+
+As a smart contract developer, I would like to wipe an amount of non-fungible tokens from an account balance through a precompile contract.
+
+## Specification
+
+### EVM Precompile extension
+
+The [IHederaTokenService.sol Solidity file](../assets/hip-206/solidity/IHederaTokenService.sol) is further extended with new types and function signatures that the precompile will respond to. 
+
+### New Solidity Types
+
+The following structs have been added to simplify the interface between Solidity and the HTS precompile: 
+
+| Name                      | Definition                                                            |
+|---------------------------|-----------------------------------------------------------------------|
+| `CustomFee`               | `(FixedFee, FractionalFee, RoyaltyFee, address)`                      |
```

Comment:
[We removed the `address` field here](https://github.com/hashgraph/hedera-smart-contracts/pull/34#discussion_r915958800).

Reply:
Removed